appcues/src/main/java/com/appcues/data/remote/DataLogcues.kt

```diff
@@ -0,0 +1,168 @@
+package com.appcues.data.remote
+
+import com.appcues.logging.Logcues
+import okhttp3.Headers
+import okhttp3.MediaType
+import okhttp3.Request
+import okhttp3.Response
+import okhttp3.internal.http.promisesBody
+import okio.Buffer
+
+internal class DataLogcues(private val logcues: Logcues) {
+
+    companion object {
+
+        private const val indent = "  "
+    }
+
+    fun error(description: String, reason: String) {
+        logcues.error("$description.\nReason: ${reason.beautify()}")
```

Comment:
I wanted to beautify() the reason if we can (makes for better visualization in debugger since if we dont do anything it will be a single liner).

then I tried to generalize how we were logging erros in the repository + interceptor and this is what I came up with.